File: Python/백트래킹/1629.py
def mul(a, b, c):
    if(b==1):
        return a%c
    ans = mul(a, b/2, c)
    ans = ans * ans % c
    if(b % 2 == 0):
        return (ans * a % c)

a, b, c = map(long, input().split())
print(mul(a, b, c))

File: Python/백트래킹/1629.c
#include <stdio.h>

int mul(long long a, long long b, long long m)
{
    if(b == 1)
        return (a % m);
    long long ans = mul(a, b/2, m);
    ans = ans * ans % m;
    if(b % 2 == 0)
        return ans;
    return (ans * a % m);
}

int main()
{
    long long a, b, m;
    scanf("%lld %lld %lld", &a, &b, &m);
    printf("%d", mul(a, b, m));
    return 0;
}